[DESU CLUB] iP

DESU-CLUB opened the pull request:
# ByteBite

>  "Can I SU this mod" - Yours truly ([source](https://medicine.nus.edu.sg/nursing/wp-content/uploads/sites/2/2019/11/Form-withdrawal-from-univ.pdf))

## **Updates**
- Finished Levels 0-6
- Created ByteBite, implementing CRUD strategies for todolist
- Added custom unicode greeting and personality

##  **Upcoming Features**
1. Find by Keyword
2. Jar release file
3. GUI

## Expected grade for this mod

- [ ] A+
- [ ] A
- [ ] A-
- [ ] B+
- [ ] B
- [ ] B-
- [ ] C
- [ ] D
- [X] See you next sem!  🤡 

---
If you're aspiring to be a janitor like me, you can reach out with a review at this [link](https://www.indeed.com/cmp/Google/reviews?fjobtitle=Janitor)

I think I'm supposed to do some code related blocks here, so have some ``Brainf*ck``code

```brainfuck
>++++++++[<+++++++++>-]<.>++++[<+++++++>-]<+.+++++++..+++.>>++++++[<+++++++>-]<+
+.------------.>++++++[<+++++++++>-]<+.<.+++.------.--------.>>>++++[<++++++++>-
]<+.
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

DESU-CLUB committed:
Refactor code to reference ByteBite

DESU-CLUB committed:
Added echo to ByteBite

DESU-CLUB committed:
Added personality to ByteBite

DESU-CLUB committed:
Add functionality to add and list commands

DESU-CLUB committed:
Add mark and unmark

DESU-CLUB committed:
Add new types of tasks

DESU-CLUB committed:
Add automated testing

DESU-CLUB committed:
Add exceptions and unit tests

DESU-CLUB committed:
Add delete function and unit tests

DESU-CLUB committed:
Add README.md

DESU-CLUB committed:
Add save and load for todolist

DESU-CLUB committed:
Merge pull request #1 from DESU-CLUB/branch-Level-7

Add save and load for todolist

DESU-CLUB committed:
Add datetime formats for ByteBite

DESU-CLUB committed:
Merge pull request #2 from DESU-CLUB/branch-Level-8

Add datetime formats for ByteBite

DESU-CLUB committed:
Refactor into distinct classes

DESU-CLUB committed:
Merge pull request #3 from DESU-CLUB/branch-A-MoreOOP

Refactor into distinct classes

DESU-CLUB committed:
Add distinct packages for refactor

DESU-CLUB committed:
Merge pull request #4 from DESU-CLUB/branch-A-Packages

Add distinct packages for refactor

DESU-CLUB committed:
Remove redundant exceptions

DESU-CLUB committed:
Merge remote-tracking branch 'origin/add-gradle-support'

DESU-CLUB committed:
Automated junit tests

DESU-CLUB committed:
Added Junit tests for some commands

DESU-CLUB committed:
Create jar file

DESU-CLUB committed:
Add javadocs for all files

DESU-CLUB committed:
Add find by keyword

DESU-CLUB committed:
Merge pull request #5 from DESU-CLUB/branch-Level-9

Branch level 9

DESU-CLUB committed:
Merge branch 'branch-Level-9' into branch-A-JavaDoc

DESU-CLUB committed:
Resolve conflict with Level-9

DESU-CLUB committed:
Merge pull request #6 from DESU-CLUB/branch-A-JavaDoc

Add javadocs for all files

DESU-CLUB committed:
Added checkstyle and fixed violations

DESU-CLUB committed:
Merge pull request #7 from DESU-CLUB/branch-A-CheckStyle

Add checkstyle and fixed violations

DESU-CLUB committed:
Add GUI for ByteBite

DESU-CLUB committed:
Merge pull request #8 from DESU-CLUB/branch-Level-10

Add GUI for ByteBite

varsha13152 commented on src/main/java/taskmanager/command/Command.java:
```diff
@@ -0,0 +1,42 @@
+     *
+     * @return true if the task list should be saved after this command, false otherwise.
+     */
+    public boolean requiresSave() {
```
According to the java coding standard, as much as possible use a prefix such as is, has, was, etc. for boolean variable/method names so that linters can automatically verify that this style rule is being followed.

varsha13152 commented on src/main/java/taskmanager/command/MarkCommand.java:
```diff
@@ -0,0 +1,64 @@
+     * @param details The task number to mark/unmark (as a string).
+     * @param markAsDone true to mark as done, false to mark as not done.
+     */
+    public MarkCommand(String details, boolean markAsDone) {
```
Instead of markAsDone consider isDone as the java coding standard document states that "As much as possible, use a prefix such as is, has, was, etc. for boolean variable/method names so that linters can automatically verify that this style rule is being followed."

DESU-CLUB replied:
Awesome! I'll modify it, thanks for the review man!

varsha13152 commented on src/main/java/taskmanager/storage/Storage.java:
```diff
@@ -0,0 +1,187 @@
+            sb.append(" | ").append(DateParser.formatForStorage(deadline.getDate()));
+        } else if (task instanceof Event) {
+            Event event = (Event) task;
+            sb.append(" | ").append(DateParser.formatForStorage(event.getStartDate()))
```
Great job! You have split the code at an appropriate point to meet the character limit

varsha13152 commented on src/main/java/taskmanager/storage/Storage.java:
```diff
@@ -0,0 +1,187 @@
+        Task task;
+        try {
+            switch (type) {
+            case "T":
```
Great! You have followed the correct style for the switch case. Usually people may make a mistake an add an additional tab to the case part of the code. Also you have managed to write such a long method without any code violations!

varsha13152 commented on src/main/java/taskmanager/task/Deadline.java:
```diff
@@ -0,0 +1,64 @@
+     *
+     * @return The date by which this task must be completed.
+     */
+    public LocalDate getDate() {
```
Great, you have even written the javadoc for getters and setters. In the java coding standard document it is mentioned that it is optional to write javadocs for getters and setter but you have gone the extra mile to do it!

varsha13152 commented on src/main/java/taskmanager/ByteBite.java:
```diff
@@ -0,0 +1,141 @@
+     * Initializes the UI, storage, and task management components.
+     * Attempts to load existing tasks from storage if available.
+     */
+    public ByteBite() {
```
In general, your code barely has any coding standard violations. In fact it was so hard to find things to comment on that I downloaded the repo and ran the checkstyle. However, there were no violations! Great job on fixing all the code violations on your own :) Keep it up!

GZTan23 commented on src/main/java/taskmanager/ByteBite.java:
```diff
@@ -0,0 +1,141 @@
+     *
+     * @param error The IOException that triggered the corrupted file handling.
+     */
+    public void handleCorruptedFile(IOException error) {
+        ui.showError("⚠️ Error detected in tasks file: " + error.getMessage());
+        try {
+            if (storage.deleteTasksFile()) {
+                ui.showMessage("Corrupted tasks file has been removed");
+                tasks = new TaskList();
+                storage.saveTasksToFile(tasks.getTaskList());
+                ui.showMessage("Created new empty tasks file");
+            } else {
+                ui.showError("❌ Unable to remove corrupted file - please check file permissions");
+            }
+        } catch (IOException e) {
+            ui.showError("❌ Critical error handling corrupted file: " + e.getMessage());
```
I like the fact that you included icons to make it more informative to the user

GZTan23 commented on src/main/java/taskmanager/task/TaskList.java:
```diff
@@ -0,0 +1,145 @@
+        tasks.add(task);
+    }
+
+    /**
+     * Deletes a task at the specified index.
+     * Uses 0-based indexing.
+     *
+     * @param index The index of the task to delete.
+     * @return The deleted task.
+     * @throws TaskNotFoundException If the index is out of bounds.
+     */
+    public Task deleteTask(int index) throws TaskNotFoundException {
+        if (index < 0 || index >= tasks.size()) {
```
Great of you to include the 0-based indexing in the JavaDocs so that the method is used correctly!

GZTan23 commented on src/main/java/taskmanager/task/TaskList.java:
```diff
@@ -0,0 +1,145 @@
+     *
+     * @return The size of the task list.
+     */
+    public int size() {
```
Maybe this should be changed to `getSize` instead so that its a verb?

GZTan23 commented on src/main/java/taskmanager/command/Command.java:
```diff
@@ -0,0 +1,42 @@
+ * All specific command types must extend this class and implement
+ * the execute method with their specific behavior.
+ */
+public abstract class Command {
```
Great job in creating an abstract `Command` class to allow for commands to be extended from, it was a stretch goal and you implemented it!

GZTan23 reviewed (commented):
Great job overall on the iP! Just like the previous reviewer, I had a hard time trying to find if there were any naming errors, which means you definitely put in a lot of thought and effort into polishing your iP. All the best for the future tasks! 😄

DESU-CLUB committed:
Add task list to gitignore

The task list changes when updated each time, and this creates a new
diff in the git repo as git tracks this file.

By adding it to gitignore, we can ensure that only intended functional
code changes prompts for a git commit.

DESU-CLUB committed:
Add assert statements for ByteBite

DESU-CLUB committed:
Merge pull request #10 from DESU-CLUB/branch-A-Assertions

Add assertions to ByteBite

DESU-CLUB committed:
Add code quality fixes

DESU-CLUB committed:
Fix code quality bug

DESU-CLUB committed:
Merge pull request #11 from DESU-CLUB/branch-A-CodeQuality

Add code quality fixes

DESU-CLUB committed:
Add tag extension

DESU-CLUB committed:
Add tag GUI extension
- Added javafx files and edited GUI java files
- Tags now show up in random colours!
- Added a new VBox so each task is its own box.

DESU-CLUB committed:
Fix visual bug

DESU-CLUB committed:
Add Ui.png and User Guide

DESU-CLUB committed:
Add Ui.png to user guide

DESU-CLUB committed:
Add style fixes

DESU-CLUB committed:
Add github workflow

DESU-CLUB committed:
Fix ci.yml

DESU-CLUB committed:
Fix ci build

DESU-CLUB committed:
Fix ci

DESU-CLUB committed:
Fix ci

DESU-CLUB committed:
Fix ci again

DESU-CLUB committed:
Fix ci again

DESU-CLUB committed:
Fix ci again

DESU-CLUB committed:
Update build.gradle

DESU-CLUB committed:
Remove gradle cache

DESU-CLUB committed:
Add GUI fix and smoke test fix

DESU-CLUB committed:
Merge branch 'master' of https://github.com/DESU-CLUB/ip

DESU-CLUB committed:
Add larger dialog avatars

DESU-CLUB committed:
Fix CI bug after merge

DESU-CLUB committed:
Fix checkstyle issue

DESU-CLUB committed:
Add new build.gradle
